website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,447 @@
+(250, 250), we need to send (250 / width, 250 / height), or (1, 0.5), as the texture coordinate. This way the size of the texture doesn't matter to the shader!
+
+<?# Info "You can use the equation ` 1/size * pixel_position ` to get the normalized coordinate for a particlar axis!" /?>
+
+Now, if you run the program, you will see just a black quad. It's because we need to bind the texture before the draw call!
+If you don't do it, the last applied texture will be used for this mesh, in our case, since we have no texture bound after the clean up, no texture is used.
+
+To solve it, go to your `OnRender` method and, below the `_gl.UseProgram()` call, first activate the texture unit that you will use:
+```c#
+_gl.ActiveTexture(TextureUnit.Texture0);
+```
+As explaned before, a texture unit is a space in memory that refers to the texture object, like a pointer. We need to first active this "pointer" to use it.
+
+After that we bind the texture again. Doing it after activating the texture unit will automatically attatch the texture at the texture unit 0, as a 2D texture.
+```c#
+_gl.BindTexture(TextureTarget.Texture2D, _texture);
+```
+With it, the texture in the texture unit 0 should be sampled for the 1° Sampler2D.
+
+
+And now when you run it (drumroll...), you can see the image being drawn inside the quad!
+
+![Quad with texture](../../../images/opengl/chapter1/lesson3/quad-with-texture.png)
+
+you can see the code final result [clicking here](../sources/1.3-final-result.html).
+
+## Transparency in OpenGL
+Well, you must have noticed the black rectangle around the texture. If you used another program to check the used texture, it's completely transparent!
+So why our render are drawing it like it's not?
+Basically because, for OpenGL eyes, it's not!
+
+Transparency is a really expensive task in computer graphics. This is not a reason for you not to use this feature, but is a reason for OpenGL don't enable it
+as default. OpenGL have different ways to handle transparency and it's expected tuser choose for that.
+
+First of all, we need to enable the blend capability. at the end of `onLoad` method, add the following line:
+```c#
+_gl.Enable(EnableCap.Blend);
+```
+Enabling blend, now OpenGL knows it should do something with the alpha value of each pixel. But what will be made?
+For define this, we use the following line:
+```c#
+_gl.BlendFunc(BlendingFactor.SrcAlpha, BlendingFactor.OneMinusSrcAlpha);
+```
+
+The `BlendFunc()` method define the function of blending of each pixel.
+In this case we are difining `BlendingFactor.SrcAlpha` as the source factor and `BlendingFactor.OneMinusSrcAlpha` as the destination blending factor;
+With this, we have a standard transparence function that will work as expected for almost all your renders.
+
+If not, you also have a lot more bledin factors to use. See the complete list in [khronos.org](https://registry.khronos.org/OpenGL-Refpages/gl4/html/glBlendFunc.xhtml).
```

Comment:
If this isn't sufficient, there are a large amount of other blending function configurations you can use, viewable at [khronos.org](https://registry.khronos.org/OpenGL-Refpages/gl4/html/glBlendFunc.xhtml).